app/models/ems_cluster.rb

```diff
@@ -107,8 +120,11 @@ def total_vcpus
   # Relationship methods
   #
 
-  alias_method :storages,               :all_storages
-  alias_method :datastores,             :all_storages    # Used by web-services to return datastores as the property name
+  has_many :storages, -> { distinct }, :through => :hosts
+  has_many :lans, -> { distinct }, :through => :hosts
+  alias_attribute :all_storages, :storages
```

Comment:
@Fryguy can we delete these `all_*` methods if we deleted the `aggregate_hardware` method?

```ruby
def aggregate_hardware(from, field, targets = nil)
      from      = from.to_s.singularize
      select    = field == :aggregate_cpu_speed ? "cpu_total_cores, cpu_speed" : field
      targets ||= send("all_#{from.pluralize}")
      hdws      = Hardware.where(from.singularize => targets).select(select)
      hdws.inject(0) { |t, hdw| t + hdw.send(field).to_i }
    end
```

Reply:
I am also curious about this.  ^

(also, pedantic, but @kbrock, your spacing is borked on that code snippet :trollface: )

Reply:
🙇 thank you @NickLaMuro for finding that an preventing me from embarrassing myself.

Reply:
before, we used all_hosts to have aggregation for the high level concepts (e.g. vms) in hierarchical trees (e.g.: service template). Here there were direct and indirect vms. and all vms.

for models where we removed the use of the mixin, these all_ virtual relations don't seem necessary anymore